ui/src/layouts/storage.rs

```diff
@@ -110,7 +117,56 @@ pub fn FilesLayout(cx: Scope<Props>) -> Element {
                                 storage_state.set(Some(storage));
                             }
                             Err(e) => {
-                                log::error!("failed to update storage with new items: {}", e);
+                                log::error!("failed to add new directory: {}", e);
+                                continue;
+                            }
+                        }
+                    }
+                    ChanCmd::OpenDirectory(directory_name) => {
+                        let (tx, rx) = oneshot::channel::<Result<Storage, warp::error::Error>>();
+                        let directory_name2 = directory_name.clone();
+
+                        if let Err(e) = warp_cmd_tx.send(WarpCmd::Constellation(
+                            ConstellationCmd::OpenDirectory {
+                                directory_name,
+                                rsp: tx,
+                            },
+                        )) {
+                            log::error!("failed to open {directory_name2} directory {}", e);
+                            return;
```

Comment:
in this scenario, `continue` rather than `return` on error. otherwise the page will stop processing commands.